bin/pt-kill

```diff
@@ -7373,17 +7373,42 @@ sub main {
 
          MATCHING_QUERY:
          foreach my $query ( @queries ) {
+            my $qinfo = ($query->{Info} || 'NULL' );
+            if( $o->get('fingerprint') && $qinfo ne 'NULL' ) {
+                     $qinfo = $qr->fingerprint($query->{'Info'});
+            }
             if ( $o->get('print') ) {
-               my $qinfo = ($query->{Info} || 'NULL' );
-               if( $o->get('fingerprint') && $qinfo ne 'NULL' ) {
-                        $qinfo = $qr->fingerprint($query->{'Info'});
-               }
-
                printf "# %s %s %d (%s %d sec) %s\n",
                   ts(time), $o->get('kill-query') ? 'KILL QUERY' : 'KILL',
                   $query->{Id}, ($query->{Command} || 'NULL'), $query->{Time},
                   $qinfo;
             }
+            if ( $o->get('log-json') ) {
+               my $reasons = join "\n", map {
+                     defined($_) ? $_ : "Unkown reason"
+                  } @{ $pl->{_reasons_for_matching}->{$query} };
+               my %jsonOutput = (
+                     ts          => ts(time),
+                     reason      => $reasons,
+                     kill_error  => $EVAL_ERROR,
+                     Id          => $query->{Id},
+                     User        => $query->{User},
+                     Host        => $query->{Host},
+                     db          => $query->{db},
+                     Command     => $query->{Command},
+                     Time        => $query->{Time},
+                     State       => $query->{State},
+                     Info        => $qinfo
+               );
+               if ( $o->get('log-json-fields') ) {
+                  my @logFieldKVs = split(",", $o->get('log-json-fields'));
+                  my %logFields = map { split(":", $_) } @logFieldKVs;
+                  %jsonOutput = (%jsonOutput, %logFields);
+               }
+               open my $fh, '>>', $o->get('log-json');
```

Comment:
Opening and closing the filehandle for each log entry seems a bit wasteful, but I couldn't see a nice way of making the filehandle variable available in the parent scope (since `--log-json` is an optional parameter). Any thoughts on a better way of doing this would be much appreciated!